### ISSUE TITLE
Auto Indent for code blocks

### ISSUE THREAD

**Issue by xyegithub** [preservim/vim-markdown#577]
Is there a method for auto indent in the code blcoks.
Like python, auto indent after `:`.

**Comment by alerque** [comment]
Are you talking about the indent level of the whole block itself or for various indents inside the block according to the content language?

**Comment by xyegithub** [comment]
> various indents inside the block according to the content language

 various indents inside the block according to the content language

**Comment by alerque** [comment]
I think that would be completely out of scope for this plugin. You can switch to a different file type and hence language mode so such niceties could be handled by the respective language plugin(s), but I don't see how this plugin could be expected to handle language specific things like that.

Maybe there is something I'm missing here, do you know of any other plugin that does this for embeded languages?

**Comment by xyegithub** [comment]
> I think that would be completely out of scope for this plugin. You can switch to a different file type and hence language mode so such niceties could be handled by the respective language plugin(s), but I don't see how this plugin could be expected to handle language specific things like that.
> 
> Maybe there is something I'm missing here, do you know of any other plugin that does this for embeded languages?

No. I am new to vim. I am trying to config my vim these days.  [`prettier`](https://prettier.io/docs/en/options.html#embedded-language-formatting) seems can format the embeding language in markdown. So, I think it will be nice, if there is any plungin can auto indent in code blocks.

**Comment by xyegithub** [comment]
> It can "format" the code as in indent and other similar things or it can just syntax highlight it?

It is a formater used by vim-autoformat or neoformat

**Comment by felixmde** [comment]
You could use one of the [Prettier Vim plug-ins](https://prettier.io/docs/en/vim.html) to run Prettier on Markdown files to utilize the embedded language feature you have mentioned.

Note that Prettier does not seem to support [Python auto-formatting](https://prettier.io/docs/en/index.html), so Prettier will not auto-format your Python code blocks. The supported languages (like JSON) should work, though.

**Comment by xyegithub** [comment]
> You could use one of the [Prettier Vim plug-ins](https://prettier.io/docs/en/vim.html) to run Prettier on Markdown files to utilize the embedded language feature you have mentioned.
> 
> Note that Prettier does not seem to support [Python auto-formatting](https://prettier.io/docs/en/index.html), so Prettier will not auto-format your Python code blocks. The supported languages (like JSON) should work, though.

Thank you for your advice!
